client/beefy/src/lib.rs

```diff
@@ -39,15 +39,39 @@ mod round;
 mod worker;
 
 pub mod notification;
-
-pub const BEEFY_PROTOCOL_NAME: &str = "/paritytech/beefy/1";
+pub use beefy_protocol_name::standard_name as protocol_standard_name;
+
+pub(crate) mod beefy_protocol_name {
+	use sc_chain_spec::ChainSpec;
+
+	const NAME: &'static str = "/beefy/1";
+	/// Old names for the notifications protocol, used for backward compatibility.
+	pub(crate) const LEGACY_NAMES: [&'static str; 1] = ["/paritytech/beefy/1"];
+
+	/// Name of the notifications protocol used by Beefy.
+	/// Must be registered towards the networking in order for Beefy to properly function.
+	pub fn standard_name<Hash: std::fmt::Display>(
+		genesis_hash: &Hash,
+		chain_spec: &Box<dyn ChainSpec>,
+	) -> std::borrow::Cow<'static, str> {
```

Comment:
What's the idea behind `Cow`? Seems like we always allocate anyway?

Reply:
Used `Cow` directly to avoid conversion since Beefy and Granpda gossip protocol names are also `Cow`.